crates/subspace-service/src/dsn/import_blocks/root_blocks.rs

```diff
@@ -179,6 +182,20 @@ impl RootBlockHandler {
         Err("Root block consensus failed: can't pass the threshold.".into())
     }
 
+    async fn validate_root_blocks(&self, peer_id: PeerId, root_blocks: &Vec<RootBlock>) -> bool {
+        // check for duplicates
+        let segment_indexes = BTreeSet::from_iter(root_blocks.iter().map(|rb| rb.segment_index()));
```

Comment:
Duplicates mean peer doesn't respond properly and should be banned. We don't need to write defensive code just to be able to process garbage if we can set the rules where garbage is not supported in the first place.

Reply:
Yes, but it ONLY checks for duplicates. It chooses to ignore which root blocks were returned (maybe none of them are those you requested), it deliberately skips check for ordering too.